datafusion/src/scalar.rs

```diff
@@ -945,7 +1018,15 @@ impl ScalarValue {
                 &DataType::LargeUtf8 => {
                     build_list!(LargeStringBuilder, LargeUtf8, values, size)
                 }
-                dt => panic!("Unexpected DataType for list {:?}", dt),
+                _ => ScalarValue::iter_to_array_list(
```

Comment:
I wonder if we could remove all the other `DataType` matches above and consolidate the code to use `iter_to_array_list` rather than `build_list!`

Though perhaps that code cleanup would lose some optimizations (aka would concat a lot of small arrays) 🤔

Reply:
My intuition was that the current specialized implementations would be more efficient, but I didn't actually benchmark that assumption.

Reply:
Leaving it this way is probably the best plan